### ISSUE TITLE
On startup current wallpaper briefly appears, then desktop is switched to dark grey

### ISSUE THREAD

**Issue by JohRest** [LGFae/swww#253]
Just recently my hyprland setup regarding wallpaper display just broke. The picture is visible briefly, then a dark gray background is being displayed. Must do on terminal a:
> swww init
DEPRECATION WARNING: `swww init` IS DEPRECATED. Call `swww-daemon` directly instead
WARNING: socket file /run/user/1000/swww.socket was not deleted when the previous daemon exited

to have it appear correctly. The same happens when I select a new wallpaper. The screen just stays dark. swww init heals it.

I'm currently using:
swww-daemon 0.9.1

My startup configuration regarding swww looks like this:

# Initialize swww
exec-once = swww init || swww-daemon --format xrgb

and: 
![image](https://github.com/LGFae/swww/assets/2835797/fd132e06-693a-47a1-8794-9e3871ba923c)

**Comment by Oglo12** [comment]
I have similar issues. Is your setup still doing this?

**Comment by preland** [comment]
I am also having this issue. Clearing cache and directly running swww-daemon allows me to see logs; when switching wallpapers it does the same behavior as mentioned before, and the daemon throws a runtime error with a buffer overflow.

**Comment by LGFae** [comment]
@preland, can you please post the error?

**Comment by Enigma1309** [comment]
try this  `swww-daemon --format xrgb &` the **&** part is important

**Comment by esalexreyes** [comment]
Same thing happens to me, while running the daemon, I got this:

```
thread 'rayon thread 22' has overflowed its stack
fatal runtime error: stack overflow
```

I was just cycling through jpgs and pngs.

**Comment by LGFae** [comment]
@esalexreyes that's issue #239. It is fixed in the latest master.

**Comment by Asteroth2018** [comment]
no its not fixed i still have the same issue even when using swww-daemon. Ive tried every combination of swww-daemon and swww init possible. The wallpaper shows and even changes once but when i want to change it second time the background goes black and after few tries of changing wallpaper it reappears. so its like random black background. I have a script to change wallpaper with keybind. and its currently
  
swww query || swww init || swww-daemon --format xrgb 

And this is the the best combination i have cause at least every few tries it shows wallpaper

**Comment by LGFae** [comment]
You are getting specifically a stack overflow error, in the latest master? Post the error and `swww --version`. If you are still getting a stack overflow in the latest master, that's a new bug.

**Comment by connorjjantzi** [comment]
I think there might be some confusion about the master. Currently I'm on version `0.9.1` but I'm not sure how I'm supposed to get the `0.9.1-master`.

**Comment by Asteroth2018** [comment]
yup sorry for the hassle installing swww-git fixed it for me aswell. I can just say when i had the issue when i removed the swww.socket from the /run/usr/1000 everything worked for a bit (like i could change wallpaper like 3 times and on the 4th try it went black) and then i would delete the socket again and the same.

**Comment by Kodlak15** [comment]
I was also experiencing a problem related to a rayon thread stack overflow. I am using NixOS with swww version 0.9.1, so if I am following correctly I don't believe the package version I am using has been patched for this issue. However, I was able to at least get my scripts working again by using all default transitions. If you are still experiencing this issue that could be a viable temporary solution for you as well.

**Comment by muni-corn** [comment]
this is fixed for me as of 0.9.4! using nixpkgs-wayland overlay on NixOS

**Comment by LGFae** [comment]
Think a bit about this, it might also be fixed by #296. Since @muni-corn is saying that is already fixed before, I will be closing this when #296 is merged. Feel free to reopen it if the problem persists.
